fl4health/clients/basic_client.py

```diff
@@ -744,14 +773,15 @@ def setup_client(self, config: Config) -> None:
         Args:
             config (Config): The config from the server.
         """
+        # Empty device cache
+        self.empty_cache()
```

Comment:
Honestly I'm not totally sure either, but nnunet does it on train start and end so I just included it as well. I assume because nnunet models are VRAM intensive that they wanted to ensure they freed as much vram as possible before training. I figured it's only called once when initializing the client and the overhead is only a few seconds at most so I didn't really see any downside to including it. The upside being just remaining consistent with nnunet, maybe there are some edge cases that I am unaware of where it is needed. I'm happy to move it into the nnunet client if you feel it does not belong in the basic client

Reply:
Hmmm...I'm not sure I know enough about how the intricacies of the caching to make a super confident call here. If nnunet does it as part of its training, I think that it makes sense for us to do it as well. However, I lean towards moving it into the nnunet client child rather than basic client for now and logging a ticket in the backlog for us to look into it more thoroughly. It feels like something the basic client might benefit from, but I just want to look into it a bit more before we commit to it. Does that make sense?

Reply:
Yep makes perfect sense